### ISSUE TITLE
Plan for ontologizing neuroNER

### ISSUE THREAD

**Issue by dosumis** [renaud/neuroNER#62]
To post on neuroNER GitHub:
- [ ] TABLE1: Full table of neuroNER terms as TSV @stripathy 
- [ ] TABLE2: Stripped down table neuroNER as TSV - uniq'd for phenotype long name & with columns for logical decomposition only.  @stripathy 

Use these tables to programatically generate new CL terms:
- PHASE 1 For each entry in the TABLE2, generate a CL term.  @dosumis
  - The short form of this term should be added to TABLE2 in a new column. 
  - Each class initially added as a subclass of neuron.
  - From TABLE1: pull NeuroLex xref (if present); PMID. 
- PHASE2: Mapping neuroNER decomposition to OWL clauses (R some C).
  - Document each mapping of decomp to OWL clause.
  - For each mapped decomposition on a term - add a corresponding SubClassOf axiom to the corresponding CL term.
  - Where all clauses of neuroNER decomposition are mapped, consider combining into a single Equivalent Class axiom.  This step requires manual review.

Phase 1: should be done ASAP.
Phase 2: map a few clauses ASAP as proof of concept.  More complete mapping is a longer term goal. 

CC @cmungall @mellybelly

**Comment by dosumis** [comment]
Example of mapping: 

![image](https://cloud.githubusercontent.com/assets/112839/16709639/a63fecf2-45e4-11e6-9725-e0d5b21964e2.png)

label "ventral hippocampus CA1 pyramidal cell"
def: "A neuron with pyramidal morphology with cell body in ventral part of hippocampus region CA1." [Cembrowski et al., Neuron 2016]  

**How far we can get given existing terms?**

SubClassOf: [pyramidal neuron](pyramidal neuron) # decomposing pyramidal as a quality = step too far?
SubClassOf: [has_soma_location](http://purl.obolibrary.org/obo/RO_0002100) some '[CA1 field of hippocampus'](http://purl.obolibrary.org/obo/UBERON_0003881)

Ventral could map to the [BSPO](https://www.ebi.ac.uk/ols/ontologies/bspo), but this should only be done if the term is used vaguely to refer to some undefined region.  Better to use a term defining the relevant subdivision if possible.  In this case ventral may map to caudal/rostral subdivision in Uberon:
![image](https://cloud.githubusercontent.com/assets/112839/16708792/04ea39e6-45cd-11e6-9800-8464a823e8ce.png)

**Comment by dosumis** [comment]
Example 2: 

![image](https://cloud.githubusercontent.com/assets/112839/16709666/ee2b6b54-45e4-11e6-8925-c8dd32ad2104.png)

neostriatum cholinergic cell
subClassOf: neuron
subClassOf: '[capable of](http://purl.obolibrary.org/obo/RO_0002215)' some '[acetylcholine secretion, neurotransmission](http://purl.obolibrary.org/obo/GO_0014055)'
subClassOf: 'has some location' some [striatum](http://purl.obolibrary.org/obo/UBERON_0002435) # is this mapping correct?  neostriatum is a synonym for striatum in Uberon, which has the note: "FMA divides 'striatum of neuraxis' into 4, neostriatum is seperate class. In NIF these are synonyms. TODO - check striatum vs corpus striatum see http://en.wikipedia.org/wiki/Striatum#History. Check [caudoputamen](http://www.ebi.ac.uk/ols/ontologies/uberon/terms?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0005383)"

**Comment by mellybelly** [comment]
RE 'ventral hippocampus CA1 pyramidal cell' I think you'd want to use BSPO for the 'ventral', I think the Uberon terms are actually referring to the other axis (though they are currently undefined TODO) but will look further. @cmungall

**Comment by dosumis** [comment]
>  I think you'd want to use BSPO for the ventral

Agreed, but only if this doesn't map to existing caudal/rostral subdivisions (may be possible if confusion between neuraxis and body axis.  Note - if we do use BSPO we need a nested class expression in CL.

**Comment by stripathy** [comment]
@dosumis , I just put together TABLE1 and TABLE2 as requested above in this commit: https://github.com/renaud/neuroNER/commit/13f61e040eafbe19d9a789f221fb73dd4b27e46c

Can we sit down during ISMB so I can walk you through the data in the tables and check and see if this is what you need for you to import these into owl and Protege on your end?

**Comment by dosumis** [comment]
Sure.  Just had a quick look.  Think we need to discuss contents of tables.  May need some tweaking. I'll email about meeting up.

**Comment by stripathy** [comment]
Updated plan for cell types to add to Cell Ontology
1. Pick a limited set of terms from neuroelectro to add to Cell Ontology (how about just the terms that are also have NeuroLex identifiers? also include CL identifiers if they're also mapped? http://www.ebi.ac.uk/ols/docs/api)
2. update spreadsheet of neuroelectro/neurolex terms to be in "wide" format (columns for brain region, morphology, etc). provide terms as names, not identifiers, for human readability. Also update brain region names from ABA to Uberon region names.
3. need to document mapping between namespaces and patterns somewhere

**Comment by tgbugs** [comment]
@stripathy I have a curator doing a phenotype decomposition of all the neurolex cells which can be seen here: https://github.com/tgbugs/nlxeol/blob/phenotypes/neuron_data_curated.csv

**Comment by cmungall** [comment]
The first thing I would do is remove all FBbts from this file. The complete
axiomatization is already there in FBbt.owl (neurolex contained a stale,
broken copy of a version of FBbt minus the complete axiomatization)

**Comment by tgbugs** [comment]
@cmungall yes, the FBbt is being purged and those entries are being skipped. I will include the FBbt.owl in nif directly in the future.
